extensions/ql-vscode/src/pure/log-summary-parser.ts

```diff
@@ -0,0 +1,42 @@
+// REVIEW: Perhaps this file should be renamed now that it also includes the 
+// data model on top of the parsing logic?
+export interface EvaluatorLogData {
+    queryCausingWork: string;
+    predicateName: string;
+    millis: number;
+    resultSize: number;
+    ra?: Pipeline[];
+}
+  
+interface Pipeline {
+    // Key: pipeline identifier; Value: array of pipeline steps 
+    pipeline: Map<string, string[]>;
+}
```

Comment:
I have simplified the model behind `Pipeline` to match what the log actually emits more closely (a map where the key is the name of the pipeline and the value is an array of pipeline steps).

Previously I had `PipelineStep` as a separate interface that only held a `body` of strings, but that additional layer of indirection seemed unnecessary. Open to other thoughts here though!

Reply:
This does not appear to match your test data, where `ra` is an object with a single field called `pipeline`, whose value is an array of strings.

Reply:
Ah, there are actually cases where `ra` contains many `pipelines` with different names, which is why I used this schema. I've added an example of this case to the `valid-summary.jsonl` file but for convenience here is the related part:

```
"ra" : {
    "base" : [
      "    {2} r1 = namespaces AND NOT namespacembrs_1#antijoin_rhs(Lhs.0)",
      "    return r1"
    ],
    "standard" : [
      "    {2} r1 = JOIN QualifiedName::Namespace::getAQualifierForMembers#f0820431#ff#prev_delta WITH QualifiedName::Namespace::getAQualifierForMembers#f0820431#ff#join_rhs#1 ON FIRST 1 OUTPUT Rhs.1, Lhs.1",
      "",
      "    {2} r2 = JOIN QualifiedName::Namespace::getAQualifierForMembers#f0820431#ff#prev_delta WITH QualifiedName::Namespace::getAQualifierForMembers#f0820431#ff#join_rhs ON FIRST 1 OUTPUT Rhs.1, (Lhs.1 ++ \"::\" ++ Rhs.2)",
      "",
      "    {2} r3 = r1 UNION r2",
      "    {2} r4 = r3 AND NOT QualifiedName::Namespace::getAQualifierForMembers#f0820431#ff#prev(Lhs.0, Lhs.1)",
      "    return r4"
    ],
    "order_500000" : [
      "    {2} r1 = JOIN QualifiedName::Namespace::getAQualifierForMembers#f0820431#ff#prev_delta WITH QualifiedName::Namespace::getAQualifierForMembers#f0820431#ff#join_rhs#1 ON FIRST 1 OUTPUT Rhs.1, Lhs.1",
      "",
      "    {2} r2 = JOIN QualifiedName::Namespace::getAQualifierForMembers#f0820431#ff#prev_delta WITH namespacembrs ON FIRST 1 OUTPUT Rhs.1, Lhs.1",
      "    {2} r3 = JOIN r2 WITH namespaces ON FIRST 1 OUTPUT Lhs.0, (Lhs.1 ++ \"::\" ++ Rhs.1)",
      "",
      "    {2} r4 = r1 UNION r3",
      "    {2} r5 = r4 AND NOT QualifiedName::Namespace::getAQualifierForMembers#f0820431#ff#prev(Lhs.0, Lhs.1)",
      "    return r5"
    ]
  },
```

From this I thought it would be most accurate for `ra` to be an array of `pipeline` objects, that have a key (the name of the pipeline such as `base`) and a value of the array of strings.

Reply:
I think a closer representation would be for `ra` to be a map/object itself, rather than an array. The keys of this map would be the (string) pipeline names, and the values of this map would be arrays of RA log strings for each pipeline.

This may also explain what you are seeing when trying to parse the JSON logs into `Pipeline[]`.

Reply:
I see; for some reason I added an extra layer of arrays in my mental model. I have renamed the interface `Pipelines` to indicate that the object contains all pipelines rather than an array of `Pipeline`s. This did indeed fix the `Pipeline[]` error. Thank you!